modules/configuration-annotation-processor/src/main/java/org/apache/ignite/internal/configuration/processor/ClassWrapper.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.configuration.processor;
+
+import static java.util.stream.Collectors.toList;
+import static javax.lang.model.element.Modifier.STATIC;
+import static org.apache.ignite.internal.configuration.processor.ConfigurationProcessorUtils.sameType;
+
+import java.lang.annotation.Annotation;
+import java.util.List;
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.TypeMirror;
+import org.apache.ignite.internal.util.Lazy;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A wrapper around a class (represented by a {@link TypeElement}) with some helper methods.
+ */
+public class ClassWrapper {
+    private final ProcessingEnvironment processingEnvironment;
+
+    private final TypeElement clazz;
+
+    private final List<VariableElement> fields;
+
+    private final Lazy<ClassWrapper> superClass = new Lazy<>(this::computeSuperClass);
```

Comment:
Please tell me why we need to use `Lazy`, maybe we can calculate it right away when initializing the object in the constructor, similar to fields?

Reply:
> maybe we can calculate it right away when initializing the object in the constructor, similar to fields?

This is not possible, because you will also recursively calculate all super classes. We also can't make this field `nullable` and initialize it during the first `get`, because `null` also represents `class doesn't have a super class`.